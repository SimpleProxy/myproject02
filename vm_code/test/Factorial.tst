// This file is part of
// File name: vm_code/test/Factorial.tst

load "../code/Factorial.asm",
output-file Factorial.out,
//compare-to Factorial.cmp,
output-list RAM[0]%D2.6.2 RAM[2]%D2.6.2;

set PC 0,
set RAM[1] 20,   // Set test arguments
set RAM[1] 20,
set RAM[2] -1;  // Ensure that program initialized product to 0
repeat 1 {
  ticktock;
}
set RAM[10] 10,   // Restore arguments in case program used them as loop counter
set RAM[10] 10,
output;

