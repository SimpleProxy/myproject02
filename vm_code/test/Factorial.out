|  RAM[0]  |  RAM[2]  |
|       1  |       0  |
